### PR TITLE
Integrate Nextest for rust

### PR DESCRIPTION
**Pull request by abhigets**
## Problem
Improve the test output and more sophisticated features like configuring timeout for test, detecting leaky tests

## Summary of changes
Add integration of nextest in dockerfile
Use cargo nextest run instead of cargo test 
Keep test run similar

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by conradludgate — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -339,17 +339,17 @@ jobs:
         run: |
           ${cov_prefix} mold -run cargo build $CARGO_FLAGS $CARGO_FEATURES --bins --tests
 
-      - name: Run cargo test
+      - name: Run nextest test
```
Comment (bayandin) [inline .github/workflows/build_and_test.yml:342]: Let's use something tool-independent, for example

```suggestion
      - name: Run Rust tests
```

File: .config/nextest.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[profile.default]
+slow-timeout = "1m"
```
Comment (bayandin) [inline .config/nextest.toml:2]:
```suggestion
slow-timeout = "1m"

```

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -339,17 +339,17 @@ jobs:
...
           # Run separate tests for real S3
           export ENABLE_REAL_S3_REMOTE_STORAGE=nonempty
-          export REMOTE_STORAGE_S3_BUCKET=neon-github-public-dev
+          export REMOTE_STORAGE_S3_BUCKET=neon-github-ci-tests
           export REMOTE_STORAGE_S3_REGION=eu-central-1
           # Avoid `$CARGO_FEATURES` since there's no `testing` feature in the e2e tests now
-          ${cov_prefix} cargo test $CARGO_FLAGS --package remote_storage --test test_real_s3
-
+          ${cov_prefix} cargo nextest run $CARGO_FLAGS -E 'package(remote_storage)' -E 'test(test_real_s3)'
+
```
Comment (bayandin) [inline .github/workflows/build_and_test.yml:352]:
```suggestion
          ${cov_prefix} cargo nextest run $CARGO_FLAGS -E 'package(remote_storage)' -E 'test(test_real_s3)'

```

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -339,17 +339,17 @@ jobs:
...
-      - name: Run cargo test
+      - name: Run nextest test
         run: |
-          ${cov_prefix} cargo test $CARGO_FLAGS $CARGO_FEATURES
+          ${cov_prefix} cargo nextest run $CARGO_FLAGS $CARGO_FEATURES
 
           # Run separate tests for real S3
           export ENABLE_REAL_S3_REMOTE_STORAGE=nonempty
-          export REMOTE_STORAGE_S3_BUCKET=neon-github-public-dev
+          export REMOTE_STORAGE_S3_BUCKET=neon-github-ci-tests
```
Comment (bayandin) [inline .github/workflows/build_and_test.yml:348]: We have these changes in the main, and they shouldn't be in the PR.

Comment (abhigets) [reply .github/workflows/build_and_test.yml:348]: I think there is something wrong with my rebase. I am currently fixing it. Hopefully it should not show up with my next push

Comment (abhigets) [reply .github/workflows/build_and_test.yml:348]: after rebasing with main this has not disappeared

**Pull request merged**